### ISSUE TITLE
Eliminate unnecessary Array manipulation when attaching with a block

### ISSUE THREAD

**Issue by adam12** [MedStack/docker-api#20]
```ruby
module Docker::Util
  def self.attach_for_tty(block, msg_stack)
    if block
      lambda do |c, r, t|
        block.call(c)
      end
    else
      messages = Docker::Messages.new

      lambda do |c, r, t|
        messages.stdout_messages << c
        messages.all_messages << c
        msg_stack.append(messages)
      end
    end
  end
end
```

**Comment by adam12** [comment]
Even better, just return the block and skip the unnecessary `lambda`.

```ruby
module Docker::Util
  def self.attach_for_tty(block, msg_stack)
    return block if block

    messages = Docker::Messages.new

    lambda do |c, r, t|
      messages.stdout_messages << c
      messages.all_messages << c
      msg_stack.append(messages)
    end
  end
end
```

**Comment by nbering** [comment]
Closed by #22
